File: 5thSemester/CellularApplicationProgramming/non_native/lib/src/model/record.dart
import 'package:hive/hive.dart';

part 'record.g.dart';

@HiveType(typeId: 0)
class Record {

  Record({
    required this.id,
    required this.rating,
    required this.opponentName,
    required this.opponentRating,
    required this.gameOutcome,
    required this.gameOpening,
    this.opponentProfile,
    this.gameTag,
    this.notes
  });

  @HiveField(0)
  late int id = 0;

  @HiveField(1)
  late int rating = 0;

  @HiveField(2)
  late String opponentName = "";
  @HiveField(3)
  late int opponentRating = 0;
  @HiveField(4)
  late String? opponentProfile = "";
  @HiveField(5)
  late String gameOutcome = "";
  @HiveField(6)
  late String gameOpening = "";
  @HiveField(7)
  late String? gameTag = "";
  @HiveField(8)
  late String? notes = "";
}

File: 5thSemester/CellularApplicationProgramming/native/app/src/main/java/com/example/chessrecords/domain/model/Record.kt
package com.example.chessrecords.domain.model

import android.icu.text.SimpleDateFormat
import android.media.Image
import java.util.Date

data class Record(
    val id: Int? = null,
    val rating: Int = 0,

    val opponentName: String = "",
    val opponentRating: Int = 0,
    val opponentProfile: String? = null,

    val gameOutcome: String = "",
    //val gameDateTime: Date = SimpleDateFormat("dd/MM/yyyy").parse("01/01/1900"),
    val gameOpening: String = "",
    val gameTag: String? = null,
    /*val gameMoves: List<String>? = null,
    val gameImage: Image? = null,*/

    val notes: String? = null
)